assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -10,28 +10,76 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { cloneBlock } from '@wordpress/blocks';
+import { useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Draggable from './draggable';
+import { isBlockAllowedOnPage } from '../../helpers';
 
-const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, index, onDragStart, onDragEnd } ) => {
+const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, index, onDragStart, onDragEnd, onNeighborDrop, getNeighborPageId } ) => {
 	const transferData = {
 		type: 'block',
 		srcIndex: index,
 		srcRootClientId: rootClientId,
 		srcClientId: clientId,
 	};
 
+	// This holds the currently highlighted element, if any
+	const hoverElement = useRef( { pageId: null, element: null, classes: [] } );
+
+	/**
+	 * Highlight neighboring page on hover. Neigboring page is given by offset.
+	 *
+	 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+	 * @param {boolean} clear  Flag used to clear highlighting without making any new highlighting
+	 */
+	const onNeighborHover = ( offset, clear = false ) => {
+		// Get neighboring page.
+		const newPageId = clear ? null : getNeighborPageId( offset );
+		const hasHighlightChanged = newPageId !== hoverElement.current.pageId;
+
+		if ( ! hasHighlightChanged ) {
+			return;
+		}
+
+		hoverElement.current.pageId = newPageId;
+
+		// Unhighlight old highlighted page.
+		if ( hoverElement.current.element ) {
+			hoverElement.current.element.classList.remove( ...hoverElement.current.classes );
+		}
+
+		// Highlight neigboring page.
+		if ( newPageId ) {
+			const isLegal = offset === 0 || isBlockAllowedOnPage( blockName, newPageId );
```

Comment:
Instead of `isLegal` why not `isAllowed`

Reply:
Fixed in 1cde15c.